### ISSUE TITLE
How can I translate this app?

### ISSUE THREAD

**Issue by amber8706** [Teknomancer/beezer#6]
I am interested in this software, but I would like to be able to localize it for my language. How can I do this?

**Comment by Teknomancer** [comment]
The language support is quite simple/rudimentary at this point.

I don't think we have support for languages other than English at the moment, but originally I wrote it with language support in mind:

Take a look at `Source/Beezer/LangStrings.h` and `Source/Beezer/LangStrings.cpp`.

If we are to add a language we will have to start from there. Afaik, we would also need some way to switch to a different language which I think might be missing in the GUI.

I think the code exists for using a different language but nothing fancy like switching languages while the app is running. So we'd need to re-start the app if we switch languages currently and an option to select a different language. Perhaps adding a drop-down in the start-up window to choose the language would be the best option.

**I don't know if this version of the source code is what Haiku ports is actually using.** This is the original source code and includes changes to port Beezer from BeOS to Haiku.

**Comment by amber8706** [comment]
What do you think about working with https://i18n.kacperkasper.pl/projects ?

**Comment by Teknomancer** [comment]
It depends on how much effort. At the moment, I might be able to check or integrate a patch (early and reasonable size please) but I don't have the time right now to take on a bigger effort.

> What do you think about working with https://i18n.kacperkasper.pl/projects ?

Sorry this requires some kind of login or allowing third party access to my GitHub account etc. Could you please point to something that does not require a login?

**Comment by Teknomancer** [comment]
Is this still relevant after the recent Haiku i18n/translation changes?

**Comment by augiedoggie** [comment]
Yes, eventually we will want to create an account at the above site so users can contribute translations using their web tool.  It should be much nicer than editing the catkeys files by hand.  We upload our en.catkeys files to the site and then we can download/import the other language catkeys files after users have submitted translations.

BTW, that site is run by one of the Haiku devs, so, it's not just some random person running the site.
